sorc/chgres_cube.fd/input_data.F90

```diff
@@ -4842,665 +5211,925 @@ subroutine read_input_sfc_grib2_file(localpet)
  
- if (.not. vgfrc_from_climo) then  
-   if (localpet == 0) then
-     print*,"- READ VEG FRACTION."
-     vname="vfrac"
-     slev=":surface:" 
-     call get_var_cond(vname,this_miss_var_method=method, this_miss_var_value=value, &
-               loc=varnum)                 
-     !! Changing these for GSD internal runs using new HRRR files
-     vname=":VEG:"
-     rc= grb2_inq(the_file, inv_file, vname,slev, data2=dummy2d)
-     
-     if (rc > 1) then
-       rc= grb2_inq(the_file, inv_file, vname,slev,'n=1105:', data2=dummy2d)
-       if (rc <= 0) then
-         rc= grb2_inq(the_file, inv_file, vname,slev,'n=1101:', data2=dummy2d)
-         if (rc <= 0) then
-           rc= grb2_inq(the_file, inv_file, vname,slev,'n=1151:', data2=dummy2d)
-           if (rc <= 0) call error_handler("COULD NOT DETERMINE VEGETATION FRACTION IN FILE.  &
-             RECORD NUMBERS MAY HAVE CHANGED. PLEASE SET VGFRC_FROM_CLIMO=.TRUE. EXITING", rc)
-         endif
-       endif
-     elseif (rc <= 0) then 
-       call error_handler("COULD NOT FIND VEGETATION FRACTION IN FILE.  &
+   if (.not. vgfrc_from_climo) then  
+
+     if (localpet == 0) then
+
+       print*,"- READ VEG FRACTION."
+
+       jdisc   = 2  ! Search for discipline - land products
+       j = 0        ! Search at beginning of file.
+       jpdtn   = 0  ! Search for product definition template number 0 - anl or fcst.
+       jpdt    = -9999  ! Initialize array of values in product definition template Sec4.
+       jpdt(1) = 0  ! Sec4/oct 10 - parameter category - veg/biomass
+       jpdt(2) = 4  ! Sec4/oct 11 - parameter number - vegetation
+       jpdt(10) = 1 ! Sec4/oct 23 - type of level - ground surface
+       unpack=.true.
+
+       call getgb2(lugb, lugi, j, jdisc, jids, jpdtn, jpdt, jgdtn, jgdt, &
+               unpack, k, gfld, rc)
+
+       if (rc /= 0 )then
+         call error_handler("COULD NOT FIND VEGETATION FRACTION IN FILE.  &
            PLEASE SET VGFRC_FROM_CLIMO=.TRUE. EXITING", rc)
-     endif
-     if(maxval(dummy2d) > 2.0) dummy2d = dummy2d / 100.0_esmf_kind_r4
-      print*,'vfrac ',maxval(dummy2d),minval(dummy2d)   
+       else
+         if (maxval(gfld%fld) > 2.0) gfld%fld = gfld%fld / 100.0
+!        print*,'vfrac ', maxval(gfld%fld),minval(gfld%fld)
+         dummy2d_8 = reshape(gfld%fld , (/i_input,j_input/))
+
+       endif
+
+     endif ! localpet 0
+
+     print*,"- CALL FieldScatter FOR INPUT GRID VEG GREENNESS."
+     call ESMF_FieldScatter(veg_greenness_input_grid,dummy2d_8, rootpet=0, rc=rc)
+     if(ESMF_logFoundError(rcToCheck=rc,msg=ESMF_LOGERR_PASSTHRU,line=__LINE__,file=__FILE__)) &
+        call error_handler("IN FieldScatter", rc)
+
    endif
 
+   if (.not. minmax_vgfrc_from_climo) then
+
+     if (localpet == 0) then
+
+       print*,"- READ MIN VEG FRACTION."
+
+       jdisc   = 2  ! Search for discipline - land products
+       j = 1105 ! grib2 file does not distinguish between the various veg
```

Comment:
I don't know if we want to address this in the current PR, or open a new one after this on is approved, but in the more recent HRRR files they actually have upgraded to using the correct wgrib2 parameter numbers for min and max veg fraction (231 and 232, respectively). Do we want to put that in as the first try for these variables here or wait and put it in a new PR? It wouldn't change any of the baseline results as the current HRRR file being used for baseline testing is old so it still needs these record number specifications. I'm honestly not sure how we haven't had an Issue submitted due to failure caused by the upgrade to the newer (and better) parameter numbers.

Reply:
I might do this in a separate PR. 

Ideally, I would like all these static fields created at the grid generation step (via sfc_climo_gen). Do we have the original greenness dataset used by HRRR? Converting it to the format used by sfc_climo_gen is not difficult.